android/src/main/java/com/oblador/keychain/KeychainModule.java

```diff
@@ -46,24 +62,34 @@ public String getName() {
     public KeychainModule(ReactApplicationContext reactContext) {
         super(reactContext);
         prefsStorage = new PrefsStorage(reactContext);
+        mReactContext = reactContext;
 
         addCipherStorageToMap(new CipherStorageFacebookConceal(reactContext));
         addCipherStorageToMap(new CipherStorageKeystoreAESCBC());
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Introducing another way of limiting ciphers by Android version might cause confusion about which one to use in the future. I think the code would be cleaner if all cipher Android version checks would be performed the same way.

So I'd suggest to either only limit them here while creating the `cipherStorageMap` and remove the redundant checks with `getMinSupportedApiLevel`, or alternatively add all ciphers to the map here and only rely on `getMinSupportedApiLevel` for the checks. I think either way is completely fine.

Reply:
> Introducing another way of limiting ciphers by Android version might cause confusion about which one to use in the future. I think the code would be cleaner if all cipher Android version checks would be performed the same way.
> 
> So I'd suggest to either only limit them here while creating the `cipherStorageMap` and remove the redundant checks with `getMinSupportedApiLevel`, or alternatively add all ciphers to the map here and only rely on `getMinSupportedApiLevel` for the checks. I think either way is completely fine.

There is a good reason for that,
the class itself has to have @TargetApi.

If I don't do that if, it won't compile.
If you have a better/cleaner solution, I am happy to implement it.